govcd/vapp.go

```diff
@@ -235,9 +235,11 @@ func (vapp *VApp) RemoveVM(vm VM) error {
 	if vapp.VApp.Tasks != nil {
 		for _, taskItem := range vapp.VApp.Tasks.Task {
 			task.Task = taskItem
-			err := task.WaitTaskCompletion()
-			if err != nil {
-				return fmt.Errorf("error performing task: %s", err)
+			if task.Task.Status != "error" && task.Task.Status != "success" {
```

Comment:
What happens when the task status is "error"? Shouldn't we handle it here?

Reply:
Could tracking vApp status directly help here instead of looking for active tasks.

Reply:
I won't touch that place any more, I did. I just spotted one case that this invalid check, as it tries to finish finished tasks. Overall here aim I guess is to waiting until all runing tasks are finished before continuing.